### ISSUE TITLE
`vue/no-unused-properties` doesn't work in JSX/TSX

### ISSUE THREAD

**Issue by Maxim-Mazurok** [vuejs/eslint-plugin-vue#1838]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 6.8.0
- **eslint-plugin-vue version:** 7.12.0
- **Node version:** v16.13.0
- **Operating System:** Windows 11

**Please show your full configuration:**

```js
module.exports = {
  root: true,
  env: {
    node: true,
  },
  plugins: ["@typescript-eslint"],
  extends: [
    "plugin:vue/essential",
    "eslint:recommended",
    "@vue/typescript/recommended",
    "plugin:@typescript-eslint/recommended",
  ],
  parserOptions: {
    ecmaVersion: 2020,
    project: "./tsconfig.eslint.json",
    extraFileExtensions: [".json"],
    ecmaFeatures: {
      jsx: true,
    },
    jsxPragma: "preserve",
  },
  rules: {
    "vue/no-unused-properties": [
      "warn",
      {
        groups: ["props", "data", "computed", "methods", "setup"],
      },
    ],
  },
};
```

**What did you do?**	

```tsx
import * as tsx from "vue-tsx-support";
import { PropType } from "vue";

export default tsx.component({
  name: "UnusedProp",
  props: {
    unusedProperty: {
      // <<< *** expected warning here ***
      type: String as PropType<string>,
      required: true,
    },
  },
  render() {
    return <h2>I have unused prop :(</h2>;
  },
});
```

tsconfig.eslint.json
```json
{
  "extends": "./tsconfig.json",
  "include": ["**/*.js", "**/*.jsx", "**/*.cjs", "**/*.vue", "**/*.ts", "**/*.tsx", "**/*.json"]
}
```

tsconfig.json
```json
{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "strict": true,
    "alwaysStrict": true,
    "jsx": "preserve",
    "jsxFactory": "VueTsxSupport",
    "importHelpers": true,
    "moduleResolution": "node",
    "experimentalDecorators": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "sourceMap": true,
    "baseUrl": ".",
    "types": ["webpack-env", "vuetify"],
    "noImplicitAny": true,
    "noUncheckedIndexedAccess": true,
    "lib": ["esnext", "dom", "dom.iterable", "scripthost"]
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"],
  "exclude": ["node_modules"]
}
```

**What did you expect to happen?**

I expected to get a warning about unused prop.

**What actually happened?**

I didn't get any warning regarding unused prop.

**Repository to reproduce this issue**

https://github.com/Maxim-Mazurok/eslint-vue-no-unused-properties-tsx-repro

**Comment by ota-meshi** [comment]
Thank you for posting this issue.

vue-tsx-support seems to be a third party library. Therefore, this project does not support vue-tsx-support.

If you really need it, ask the vue-tsx-support team to develop the eslint-plugin.
